### PR TITLE
Add credhub support

### PR DESCRIPTION
**Pull request by dennisjbell**
Adds a credhub bash helper function for hooks, and the ability to specify that an environment uses credhub (so don't do the vault validations).

The credhub helper requires an environment that provides the connection information via exodus data.  The latest BOSH genesis kit contains these values.  By default, the bosh environment name will be used, but can be overrode by using `genesis.credhub-exodus-env`

### CODE REVIEW

File: lib/Genesis/Kit.pm [hunk 1]
```diff
@@ -242,6 +253,13 @@ sub metadata {
 	return $self->{__metadata};
 }
 
+# }}}
+# uses_credhub - does this kit use credhub instead of vault {{{
+sub uses_credhub {
+	my ($self) = @_;
+	return defined($self->metadata->{secret_store}) && $self->metadata->{uses_credhub} eq "credhub";
```
Comment (krutten) [inline lib/Genesis/Kit.pm]: second value should also be "uses_credhub"

File: lib/Genesis/Kit/Compiler.pm [hunk 2]
```diff
@@ -68,7 +68,18 @@ sub validate {
 		}
 
 		# check for errant top-level keys - params, subkits and features have been discontinued.
-		my @valid_keys = qw/name version description code docs author authors genesis_version_min certificates credentials/;
+		my @valid_keys = qw/name version description code docs author authors genesis_version_min secrets_store/;
+		if ($meta->{secrets_store}) {
+			if ($meta->{secrets_store} eq "credhub") {
+				#no-op: valid, no further keys
+			} elsif ($meta->{secrets_store} eq "credhub") {
```
Comment (krutten) [inline lib/Genesis/Kit/Compiler.pm]: credhub should vault

File: lib/Genesis/Kit.pm (hunk 1 above)
Comment (dennisjbell) [reply lib/Genesis/Kit.pm]: fixed

File: lib/Genesis/Kit/Compiler.pm (hunk 2 above)
Comment (dennisjbell) [reply lib/Genesis/Kit/Compiler.pm]: fixed

**Pull request merged**